### PR TITLE
Propagate +adios2 to paraview for Data and Vis SDK

### PR DESCRIPTION
**Pull request by kwryankrattiger**
Depends on #26728 and #26175

@chuckatkins

### CODE REVIEW

File: var/spack/repos/builtin/packages/ecp-data-vis-sdk/package.py
Comment (kwryankrattiger) [inline var/spack/repos/builtin/packages/ecp-data-vis-sdk/package.py]: Only change is here.

**Comment by vicentebolea** [comment]
@kwryankrattiger  looks like a good refactor, few things:

- What does DAV stands for?
- How about making a more verbose PR description
- It seems to be rebased on the top of other contributions, are those from the PR referenced in the description of this issue?

**Comment by kwryankrattiger** [comment]
@vicentebolea 

>     * What does DAV stands for?
Data and Visualization, @chuckatkins recommended the acronym for use in the SDK package.

>     * It seems to be rebased on the top of other contributions, are those from the PR referenced in the description of this issue?
There is one by me that is the main refactor the rest of them are from a rebase onto `spack/develop`, I think I may have rebased things in the wrong order, I will re-propagate all of the rebases for this branch.

**Review by vicentebolea — DISMISSED** [review]
LGTM

**Comment by vicentebolea** [comment]
+1 after fixing style check

**Review by chuckatkins — DISMISSED** [review]

**Review by vicentebolea — DISMISSED** [review]

**Comment by haampie** [comment]
@spackbot run pipeline

**Comment by kwryankrattiger** [comment]
@spackbot run pipeline

**Comment by scottwittenburg** [comment]
The `hpx` package failed on gitlab with:

```
/builds/spack/spack/var/spack/repos/builtin/packages/hpx/package.py:166, in instrumentation_args:
        165    def instrumentation_args(self):
  >>    166        for value in self.variants['instrumentation'].values:
        167            if value == 'none':
        168                continue
        169
```

I think that was fixed in [1dcabdbc8d](https://github.com/spack/spack/commit/1dcabdbc8d7aa71348a262e659a528242b1cd25e), in which case this PR needs a rebase.

**Review by chuckatkins — DISMISSED** [review]

**Comment by kwryankrattiger** [comment]
@adamjstewart Did you intend to block this merge? This is ready now after rebasing on #27784 merged which fixed enabling ADIOS2 with ParaView.

**Comment by adamjstewart** [comment]
Hmm, I don't remember dismissing @chuckatkins review. Usually when I do this it's because someone requested changes and the changes were added but the person never re-reviewed. Reviews also get dismissed when you rebase, maybe that's what happened? Anyway, no problems with this being merged from my side.

**Review by chuckatkins — APPROVED** [review]

**Pull request merged**
